File: src/client/java/com/myudog/myulib/client/api/ui/component/WidgetStateComponent.java
package com.myudog.myulib.client.api.ui.component;

import com.myudog.myulib.api.core.ecs.IComponent;

public class WidgetStateComponent implements IComponent {
    public boolean visible = true;
    public boolean enabled = true;
    public boolean hovered;
    public boolean pressed;
    public boolean focused;
}

File: src/client/kotlin/com/myudog/myulib/client/api/ui/component/WidgetStateComponent.kt
package com.myudog.myulib.client.api.ui.component

import com.myudog.myulib.api.ecs.Component

data class WidgetStateComponent(
    var isVisible: Boolean = true,
    var isEnabled: Boolean = true,
    var isHovered: Boolean = false,
    var isFocused: Boolean = false,
) : Component